service/namespace.go

```diff
@@ -11,7 +14,12 @@ const resultChannel string = "Result"
 
 // namespace returns the namespace of the service.
 func (service *Service) namespace() []string {
-	return []string{service.Hash}
+	h, err := hex.DecodeString(service.Hash)
+	if err != nil {
+		panic(err)
```

Comment:
I don't really like the idea to have a panic here (or even the fatal)
```suggestion
		logrus.Fatal(err)
```

Reply:
It's just temporary before refactor. It will be gone.